nautobot/dcim/tests/test_schema.py

```diff
@@ -0,0 +1,84 @@
+from nautobot.core.models.utils import get_all_concrete_models
+from nautobot.core.testing.schema import OpenAPISchemaTestCases
+from nautobot.dcim.models import CableTermination, PathEndpoint
+
+
+class DCIMOpenAPISchemaTestCase(OpenAPISchemaTestCases.BaseSchemaTestCase):
```

Comment:
I want to be clear in that I think this is awesome.

What I don't like is how there is so much hard-coding of object names and nested fields relationships. I wish there were a way we could abstract this somehow.

If we were, for example, to manually add schema tests across the board, it would be incredibly tedious.

I know you added these as sample tests and it seems reasonable that this lays the foundation for implementing generic tests based on expectations provided by the way things are marked up using schema decorators?

Reply:
As discussed, I'll take a quick stab at refining this a bit to reduce the repetition.

Reply:
Or spend a couple hours on it, whichever. 😅 Please take a look at the updated tests.